tests/test_testserver.py

```diff
@@ -141,6 +228,44 @@ def test_request_recovery_with_bigger_timeout(self):
 
         assert server.handler_results[0] == data
 
+    def test_request_recovery_with_bigger_timeout2(self):
+        """a biggest timeout can be specified"""
+        server = Server.basic_response_server(request_timeout=3)
+        data = b"bananadine"
+
+        with server as address:
+            sock = socket.socket()
+            sock.connect(address)
+            time.sleep(1.5)
+            sock.sendall(data)
+            sock.close()
+            
+    def test_request_recovery_with_bigger_timeout3(self):
+        """a biggest timeout can be specified"""
+        server = Server.basic_response_server(request_timeout=3)
+        data = b"bananadine"
+
+        with server as address:
+            sock = socket.socket()
+            sock.connect(address)
+            time.sleep(3)
+            sock.sendall(data)
+            sock.close()
+
+    def test_request_recovery_with_bigger_timeout4(self):
+        """a biggest timeout can be specified"""
+        server = Server.basic_response_server(request_timeout=3)
+        data = b"bananadine"
+
+        with server as address:
+            sock = socket.socket()
+            sock.connect(address)
+            time.sleep(20)
+            sock.sendall(data)
+            sock.close()
```

Comment:
**Suggestion:** The test methods `test_request_recovery_with_bigger_timeout2`, `test_request_recovery_with_bigger_timeout3`, and `test_request_recovery_with_bigger_timeout4` are almost identical. Consider parameterizing these tests to avoid code duplication.
```suggestion
    @pytest.mark.parametrize("sleep_time", [1.5, 3, 20])
    def test_request_recovery_with_bigger_timeout(self, sleep_time):
        """a biggest timeout can be specified"""
        server = Server.basic_response_server(request_timeout=3)
        data = b"bananadine"

        with server as address:
            sock = socket.socket()
            sock.connect(address)
            time.sleep(sleep_time)
            sock.sendall(data)
            sock.close()
```